### PR TITLE
docs(hashgraph): initial draft

### PR DESCRIPTION
**Pull request by h3lio5**

### CODE REVIEW

File: docs/get-started/hashgraph.md [hunk 1]
```diff
@@ -6,30 +6,59 @@ sidebar_position: 3
...
+- u is the operation
+- D is the set of hashed vertices that are its causal dependencies
 
-Here is an example of a CRO's hash graph:
+This structure ensures that if operation u2 reports u1 as its causal dependency, then u2 must have happened after u1. By using a deterministic conflict resolution heuristic defined in the CRO, the union (or merging in topology's parlance) of hashgraphs is order-agnostic -- meaning the final state is the same even if different replicas merge hashgraphs from their replica peers in different orders.
 
 <div align="center">
-    ![alt text](/img/hash_graph.png)
+  
+![Hash graph of a CRO](/img/hash_graph.png)
```
Comment (guiltygyoza) [inline docs/get-started/hashgraph.md:30]: let's make our own drawings

File: docs/get-started/hashgraph.md [hunk 2]
```diff
@@ -6,30 +6,59 @@ sidebar_position: 3
...
+</div>
+
+In the example above, vertex `V7` should contain `(u7, {h(V4), h(V5)})`. The set of vertices `{V6, V7, V8}` constitutes the frontier, which are vertices whose operations are currently not dependencies of any other operation. The next vertex, `V9`, should have `{h(V6), h(V7), h(V8)}` as its causal dependencies.
 
-    **Figure 1:** Hash graph of a CRO.
 
+# HashGraph Merge
+When two nodes synchronize their operation histories of the same CRO, they effectively merge their hash graphs. The merge operation performs the union of hashgraphs. Merging might require resolving concurrent operations, but this is out of scope for this article (check out the [concurrency](./concurrency.md) and [conflict](./conflict.md) sections for more details).
+
+Notably, this approach is immune to sybil attacks, allowing CROs to tolerate many sybil actors in the network.
```
Comment (guiltygyoza) [inline docs/get-started/hashgraph.md:41]: we'll need to explain this in much simpler terms while being precise. "many sybil actors" so precisely how many? are we still subject to the f / 3f+1 limitation?

File: docs/get-started/hashgraph.md [hunk 3]
```diff
@@ -6,30 +6,59 @@ sidebar_position: 3
...
 </div>
 
-In the example above, the vertex _V7_ should contain (_u7_, \{_h(V4)_, _h(V5)_}). The set of vertices \{_V6_, _V7_, _V8_} consists on the _frontier_, which are the vertices whose operations are currently not the dependencies of any other operation. The next vertice, _V9_, should have \{_h(V6)_, _h(V7)_, _h(V8)_} as its causal dependencies.
+# Eventual Consistency
+
+Eventual consistency is a fundamental property of CROs that guarantees all replicas will eventually reach the same state, regardless of the order in which they receive and process operations. This is achieved through several key mechanisms:
+
+1. **Causal History Preservation**: The hashgraph structure maintains the complete causal history of operations through its (u, D) vertex structure, ensuring that causally related operations are always processed in the correct order across all replicas.
+
+2. **Deterministic Conflict Resolution**: When concurrent operations occur, the CRO's conflict resolution rules ensure that all replicas resolve conflicts in exactly the same way, regardless of the order in which they receive the operations.
```
Comment (guiltygyoza) [inline docs/get-started/hashgraph.md:55]: don't introduce it here yet. "conflict" is introduced in the next section. it's a new term for most devs

**Review by guiltygyoza — CHANGES_REQUESTED** [review]
requested changes / reconsiderations

File: docs/get-started/hashgraph.md (hunk 3 above)
Comment (h3lio5) [reply docs/get-started/hashgraph.md:55]: can I make `Deterministic Conflict Resolution` embed a hyperlink to the conflicts page? Wanted to keep conflicts here just for sake of completeness of explaining how eventual consistency is guaranteed.

File: docs/get-started/hashgraph.md (hunk 1 above)
Comment (h3lio5) [reply docs/get-started/hashgraph.md:30]: ok, but i took this from our whitepaper. Perhaps you meant the hash graph image?

File: docs/get-started/hashgraph.md (hunk 2 above)
Comment (h3lio5) [reply docs/get-started/hashgraph.md:41]: i think it's better to leave out this detail until the finality design is complete? Don't think we can be super precise about bft rn.

File: docs/get-started/hashgraph.md [hunk 4]
```diff
@@ -6,30 +6,56 @@ sidebar_position: 3
...
+  
+![Hash graph of a CRO](/img/hashgraph_new.png)
 
-    **Figure 1:** Hash graph of a CRO.
+**Figure 1:** Hash graph of a CRO.
+</div>
+
+
+# HashGraph Merge
+When two nodes synchronize their operation histories of the same CRO, they effectively merge their hash graphs. The merge operation performs the union of hashgraphs. Merging might require resolving concurrent operations, but this is out of scope for this article (check out the [concurrency](./concurrency.md) and [conflict](./conflict.md) sections for more details).
```
Comment (guiltygyoza) [inline docs/get-started/hashgraph.md:37]: Merging does not require resolving concurrent operations

Comment (guiltygyoza) [reply docs/get-started/hashgraph.md:37]: also it's much clearer to add a short paragraph explaining the following graph

File: docs/get-started/hashgraph.md (hunk 2 above)
Comment (guiltygyoza) [reply docs/get-started/hashgraph.md]: yes we can. see https://arxiv.org/pdf/2012.00472 figure 1

Comment (guiltygyoza) [reply docs/get-started/hashgraph.md]: we don't have finality in the protocol yet. but hashgraph already provides certain bft properties and it's important to explain what's provided

File: docs/get-started/hashgraph.md (hunk 3 above)
Comment (guiltygyoza) [reply docs/get-started/hashgraph.md]: yes ofc

**Review by guiltygyoza — CHANGES_REQUESTED** [review]
requested changes

**Review by guiltygyoza — APPROVED** [review]
the sybil resistant part needs clarification, to be done in a separate PR. lgtm

**Pull request merged**